File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetVideoListUseCaseImpl.java
package com.hwonchul.movie.domain.usecase.details;

import androidx.annotation.NonNull;

import com.hwonchul.movie.domain.model.Video;
import com.hwonchul.movie.domain.repository.VideoRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Flowable;

public class GetVideoListUseCaseImpl implements GetVideoListUseCase {
    private final VideoRepository repository;

    @Inject
    public GetVideoListUseCaseImpl(@NonNull VideoRepository repository) {
        this.repository = repository;
    }

    @Override
    public Flowable<List<Video>> invoke(int movieId) {
        return repository.getAllVideosByMovieId(movieId);
    }
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/GetVideoListUseCaseImpl.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.model.Video
import com.hwonchul.movie.domain.repository.VideoRepository
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.catch
import kotlinx.coroutines.flow.map
import javax.inject.Inject

class GetVideoListUseCaseImpl @Inject constructor(
    private val repository: VideoRepository
) : GetVideoListUseCase {

    override fun invoke(movieId: Int): Flow<Result<List<Video>>> {
        return repository.getAllVideosByMovieId(movieId)
            .map { Result.success(it) }
            .catch { emit(Result.failure(it)) }
    }
}